pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -430,3 +319,122 @@ func pluginsNeeded(plugins *config.Plugins) map[string]struct{} {
 
 	return pgMap
 }
+
+func (f *framework) appendPlugins(plugins *config.Plugins) error {
```

Comment:
I am not sure if the new code is better than the existing one. I feel that the existing one is more readable, but it could just be a matter of preference. The new code with the `extensionPoints` slice and `appendPluginFn` and the `switch/case` inside the function looks more complex than the existing code. Yet, it has not removed the problem of having repeated code blocks as the code inside each `case` statement is repeated (with small differences).
I don't have a better suggestion to improve this PR. I wonder if we should make this change at all.